File: Bangji/0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        groups = {}
        for string in strs:
            key = "".join(sorted(string))
            if key not in groups:
                groups[key] = [string]
            else:
                groups[key].append(string)
            
        return groups.values()
                

File: Kobe/0049-group-anagrams/0049-group-anagrams.ts
function groupAnagrams(strs: string[]): string[][] {
    const strMap = new Map();
    const res = [];

    strs.forEach((str) => {
        // 같은 문자열 구조는 키를 공유한다. 배열 분해 후 정렬 후 재조합
        const strKey = str.split('').sort().join('');
        
        if (strMap.get(strKey)) {
            strMap.get(strKey).push(str);
        } else {
            strMap.set(strKey, [str])
        }
    })
    
    strMap.forEach((value) => {
        res.push(value);
    })
    
    return res;
};